File: src/lib/commands/settings-commands.ts
import { invoke } from "@tauri-apps/api/core";
import type { Settings } from "../../types/settings";

export function loadSettings(): Promise<Settings> {
  return invoke<Settings>("load_settings");
}

export function saveSettings(settings: Settings): Promise<void> {
  return invoke("save_settings", { settings });
}

File: src-tauri/src/commands/settings_commands.rs
use crate::error::AppError;
use crate::settings::{persist, Settings};

#[tauri::command]
pub fn load_settings() -> Result<Settings, AppError> {
    let paths = persist::SettingsPaths::default_paths();
    Ok(persist::load(&paths))
}

#[tauri::command]
pub fn save_settings(settings: Settings) -> Result<(), AppError> {
    let paths = persist::SettingsPaths::default_paths();
    persist::save(&settings, &paths)
}
